lib/Twig/TokenParser/CacheParser.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Twig\TokenParser;
+
+use Countable;
+use LogicException;
+use Pimcore\Twig\Node\CacheNode;
+use Pimcore\ValueObject\Collection\ArrayOfStrings;
+use Twig\Error\SyntaxError;
+use Twig\Node\Expression\ArrayExpression;
+use Twig\Node\Node;
+use Twig\Token;
+use Twig\TokenParser\AbstractTokenParser;
+use Twig\TokenStream;
+use ValueError;
+use function count;
+use function is_int;
+use function is_null;
+use function sprintf;
+
+/**
+ * @internal
+ */
+class CacheParser extends AbstractTokenParser
+{
+    public function parse(Token $token): Node
```

Comment:
We should also add the throw tag here? 
```suggestion
    /**
     * @throws SyntaxError
     */
    public function parse(Token $token): Node
```

Reply:
This is already added in the interface (see https://github.com/twigphp/Twig/blob/3.x/src/TokenParser/TokenParserInterface.php#L36). Also the official twig extensions don't re-add it there, so I'd say we can skip it...